### PR TITLE
TASK: Add type declaration to Neos cache classes

### PR DESCRIPTION
**Pull request by daniellienert**
- Adds missing type declarations, especially void types.
- The remaining methods are implementing the untyped PSR interface

### CODE REVIEW

**Comment by daniellienert** [comment]
@albe: Again psalm delivered great suggestions and found some possible errors!

**Comment by daniellienert** [comment]
Are there any objections to merge this?

File: Neos.Cache/Classes/Backend/MultiBackend.php [hunk 1]
```diff
@@ -221,10 +223,10 @@ public function collectGarbage()
     }
 
     /**
-     * @param array $backendConfigurations
+     * @param mixed[] $backendConfigurations
```
Comment (albe) [inline Neos.Cache/Classes/Backend/MultiBackend.php]: Huh, that's an odd way to denote `array` 🤨

Comment (kdambekalns) [reply Neos.Cache/Classes/Backend/MultiBackend.php]: Yup, wouldn't do that. I get `string[]` to make clear what is expected, but `mixed[]` is in no way better than `array` IMHO.

Plus: currently a type like `string[]` breaks Doctrine, so using this in entities would be bad. So let's keep that notation out for now, please. Avoids confusion.

Comment (kdambekalns) [reply Neos.Cache/Classes/Backend/MultiBackend.php]: Then again… `string[]` really is helpful. (But `mixed[]`… not so much.)

**Comment by daniellienert** [comment]
Got it 😉 I’ll take care. Analyzed the classes with phpcs, and in my config it complains about array annotations

**Pull request merged**